vital/types/detected_object_type.h

```diff
@@ -129,6 +141,21 @@ class VITAL_EXPORT detected_object_type
    */
   double score( const std::string& class_name ) const;
 
+  /**
+   * @brief Get max class name.
+   *
+   * This method returns the most likely class for this object.
+   *
+   * If there are no scores associated with this detection, then an
+   * exception is thrown
+   *
+   * @param[out] max_name Class name with the maximum score.
+   *
+   * @throws std::runtime_error If no scores are associated with this
+   *                            detection.
+   */
+  void get_most_likely( std::string& max_name ) const;
+
```

Comment:
This would be better if the signature is as follows:

    std::string get_most_likely_class() const;

This approach gets away from returning a value through a parameter.
Could add get_most_likely_score() for completeness, but not required.